ngraph/core/src/pass/low_latency.cpp

```diff
@@ -90,3 +105,178 @@ ngraph::pass::LowLatency::LowLatency()
     register_matcher(m, callback);
 }
+NGRAPH_SUPPRESS_DEPRECATED_END
+
+void UnrollSingleIteration(const shared_ptr<op::util::SubGraphOp>& sub_graph_op,
+                           const shared_ptr<Function>& outer_f)
+{
+    using namespace opset7;
+
+    const auto& params = sub_graph_op->get_function()->get_parameters();
+    const auto& results = sub_graph_op->get_function()->get_results();
+
+    // before: Layer1 -> TI [input -> bodyParameter -> Layer2 -> ...]
+    // after:  Layer1 -> Layer2 ->...
+    for (const auto& in : sub_graph_op->get_input_descriptions())
+    {
+        const auto& connect_to = sub_graph_op->get_input_source_output(in->m_input_index);
+        for (auto& output : params.at(in->m_body_parameter_index)->outputs())
+        {
+            output.replace(connect_to);
+        }
+    }
+
+    // before: TI [...-> Layer1 -> Result -> output] -> Layer2 -> ...
+    // after:  ...-> Layer1 -> Layer2 -> ...
+    NodeVector new_ops;
+    for (const auto& out : sub_graph_op->get_output_descriptions())
+    {
+        const auto& connect_to = results.at(out->m_body_value_index)->get_input_source_output(0);
+        for (auto& input_to : sub_graph_op->output(out->m_output_index).get_target_inputs())
+        {
+            // create IE output name
+            std::string out_name = sub_graph_op->get_friendly_name();
+            if (sub_graph_op->get_output_size() != 1)
+                out_name += "." + std::to_string(out->m_output_index);
+
+            // IECompatibility: insert identity (Unsqueeze + Squeeze) to store the TensorIterator
+            // output names
+            auto axis_1 = Constant::create(ngraph::element::i64, ngraph::Shape{1}, {1});
+            auto identity_1 = std::make_shared<Unsqueeze>(connect_to, axis_1);
+            auto identity_2 = std::make_shared<Squeeze>(identity_1, axis_1);
+            identity_2->set_friendly_name(out_name);
+            new_ops.push_back(identity_1);
+            new_ops.push_back(identity_2);
+
+            input_to.replace_source_output(identity_2);
+        }
+    }
+    outer_f->add_sinks(sub_graph_op->get_function()->get_sinks());
+    ngraph::copy_runtime_info(sub_graph_op, sub_graph_op->get_function()->get_ops());
+    ngraph::copy_runtime_info(sub_graph_op, new_ops);
+}
+
+Output<Node> create_init_subgraph(const shared_ptr<op::util::SubGraphOp>& sub_graph_op,
+                                  const Output<Node>& in_node)
+{
+    using namespace opset7;
+
+    auto const_zero = make_shared<Constant>(in_node.get_element_type(), Shape{1}, 0);
+    auto shape_of = make_shared<ShapeOf>(in_node);
+    auto broadcast = make_shared<Broadcast>(const_zero, shape_of);
+    copy_runtime_info(sub_graph_op, {const_zero, shape_of, broadcast});
+    return broadcast->output(0);
+}
+
+bool pass::LowLatency2::run_on_function(shared_ptr<Function> f)
+{
+    using namespace opset7;
+
+    SinkVector assigns;
+    for (const auto& op : f->get_ops())
```

Comment:
I suggest to use `get_ordered_ops` just to guaranty that sinks will be added in the same order from run to run.

Reply:
done